data/registers/otgfs_v1.yaml

```diff
@@ -1,140 +1,297 @@
-      fieldset: OTG_FS_GRXSTSP_Device
-    - name: OTG_FS_GRXSTSP_Host
-      description: OTG status read and pop register (Host mode)
+      fieldset: GRXSTS_Device
+    - name: GRXSTSP_Host
+      description: Status read and pop (Host mode)
       byte_offset: 32
       access: Read
-      fieldset: OTG_FS_GRXSTSP_Host
-    - name: OTG_FS_GRXFSIZ
-      description: OTG_FS Receive FIFO size register (OTG_FS_GRXFSIZ)
+      fieldset: GRXSTS_Host
+    - name: GRXFSIZ
+      description: Receive FIFO size register
       byte_offset: 36
-      fieldset: OTG_FS_GRXFSIZ
-    - name: OTG_FS_DIEPTXF0_Device
-      description: OTG_FS Endpoint 0 Transmit FIFO size
+      fieldset: GRXFSIZ
+    - name: DIEPTXF0
+      description: Non-periodic transmit FIFO size register (Device mode)
       byte_offset: 40
-      fieldset: OTG_FS_DIEPTXF0_Device
-    - name: OTG_FS_HNPTXFSIZ_Host
-      description: OTG_FS Host non-periodic transmit FIFO size register
+      fieldset: DIEPTXF0
+    - name: HNPTXFSIZ
+      description: Non-periodic transmit FIFO size register (Host mode)
       byte_offset: 40
-      fieldset: OTG_FS_HNPTXFSIZ_Host
-    - name: OTG_FS_HNPTXSTS
-      description: OTG_FS non-periodic transmit FIFO/queue status register (OTG_FS_GNPTXSTS)
+      fieldset: HNPTXFSIZ
+    - name: GNPTXSTS
+      description: Non-periodic transmit FIFO/queue status register
       byte_offset: 44
       access: Read
-      fieldset: OTG_FS_HNPTXSTS
-    - name: OTG_FS_GI2CCTL
-      description: OTG I2C access register
-      byte_offset: 48
-      fieldset: OTG_FS_GI2CCTL
-    - name: OTG_FS_GCCFG
-      description: OTG_FS general core configuration register (OTG_FS_GCCFG)
+      fieldset: GNPTXSTS
+    - name: GCCFG
+      description: General core configuration register
       byte_offset: 56
-      fieldset: OTG_FS_GCCFG
-    - name: OTG_FS_CID
-      description: core ID register
+      fieldset: GCCFG
+    - name: CID
+      description: Core ID register
       byte_offset: 60
-      fieldset: OTG_FS_CID
-    - name: OTG_FS_GLPMCFG
-      description: OTG core LPM configuration register
-      byte_offset: 84
-      fieldset: OTG_FS_GLPMCFG
-    - name: OTG_FS_GPWRDN
-      description: OTG power down register
-      byte_offset: 88
-      fieldset: OTG_FS_GPWRDN
-    - name: OTG_FS_GADPCTL
-      description: "OTG ADP timer, control and status register"
-      byte_offset: 96
-      fieldset: OTG_FS_GADPCTL
-    - name: OTG_FS_HPTXFSIZ
-      description: OTG_FS Host periodic transmit FIFO size register (OTG_FS_HPTXFSIZ)
+      fieldset: CID
+    - name: HPTXFSIZ
+      description: Host periodic transmit FIFO size register
       byte_offset: 256
-      fieldset: OTG_FS_HPTXFSIZ
-    - name: OTG_FS_DIEPTXF1
-      description: OTG_FS device IN endpoint transmit FIFO size register (OTG_FS_DIEPTXF1)
+      fieldset: HPTXFSIZ
+    - name: DIEPTXF
+      description: Device IN endpoint transmit FIFO size register
+      array:
+        len: 3
```

Comment:
do all chips have max 3 endpoints? iirc some have more, we should set this to the max, the HAL should know the actual max endpoint count.

Reply:
From what I see in the extracted yamls (./d extract-all OTG_FS_{GLOBAL,HOST,DEVICE,PWRCLK}):
* The f101 f102 f103 f107 f215 f217 f412 f413 l4x6 families: all have only 3 endpoints
* The f469 family has 5 endpoints. Its datasheet shows High-Speed support, not just Full-Speed. From a quick glance I don't see extra registers in the yaml.
* The f730 f745 f750 f765 f7x2 f7x3 f7x6 f7x7 f7x9 families: 5 endpoints, but they also have a bunch of additional registers like GI2CCTL GLPMCFG GPWRDN GADPCTL. So that should be a separate yaml definition
Overall, I suspect that the fanciers cores with more endpoints have extra features, and deserve their own v2 yaml.
At this point, it's probably not harmful to keep the endpoint at 3. The fancier cores will be limited, yes. But to give them justice we'll have to do better than just bumping the number of endpoints.